jena-arq/src/main/java/org/apache/jena/sparql/expr/aggregate/AggModeDistinct.java

```diff
@@ -0,0 +1,140 @@
+    @Override
+    public Aggregator copy(ExprList expr) { return new AggModeDistinct(expr.get(0)) ; }
+
+    private static final NodeValue noValuesToMode = NodeValue.nvZERO ; 
+
+    @Override
+    public Accumulator createAccumulator()
+    { 
+        return new AccModeDistinct(getExpr()) ;
+    }
+
+    @Override
+    public Node getValueEmpty()     { return NodeValue.toNode(noValuesToMode) ; } 
+
+    @Override
+    public int hashCode()   {
+        return HC_AggModeDistinct ^ getExprList().hashCode() ;
+    }
+
+    @Override
+    public boolean equals(Aggregator other, boolean bySyntax) {
+        if ( other == null ) return false ;
+        if ( this == other ) return true ;
+        if ( ! ( other instanceof AggModeDistinct ) ) return false ;
+        AggModeDistinct a = (AggModeDistinct)other ;
+        return exprList.equals(a.exprList, bySyntax) ;
+    }
+
+    
+    // ---- Accumulator
+    class AccModeDistinct extends AccumulatorExpr
+    {
+        // Non-empty case but still can be nothing because the expression may be undefined.
+        private NodeValue total = noValuesToMode ;
+        private int count = 0 ;
+        ArrayList<NodeValue> collection=new ArrayList<NodeValue>(); 
+        
+        public AccModeDistinct(Expr expr) { super(expr, true) ; }
+
+        @Override
+        protected void accumulate(NodeValue nv, Binding binding, FunctionEnv functionEnv)
+        { 
+			log.debug("mode {}", nv);
+
+            if ( nv.isNumber() )
+            {
+                count++ ;
+                collection.add(nv);
+            }
+            else
+                throw new ExprEvalException("mode: not a number: "+nv) ;
+
+            log.debug("mode count {}", count);
+        }
+
+        @Override
+        public NodeValue getAccValue()
+        {
+            double mode= Double.NaN;
+            if ( count == 0 ) return noValuesToMode ;
+            if ( super.errorCount != 0 )
+                return null ;
+            
+            int indexsize = collection.size();
+            double[] arrDouble = new double[indexsize];
+            for(int i=0; i<indexsize; i++){
+            	arrDouble[i] = collection.get(i).getDouble();	
+            }
+
+    	    HashMap<Double,Integer> amode = new HashMap<Double,Integer>();
+    	    int max  = 0;
+
+    	    for(int i = 0; i < arrDouble.length; i++) {
+    	        if (amode.get(arrDouble[i]) != null) {
+    	            int count = amode.get(arrDouble[i]);
+    	            count++;
+    	            amode.put(arrDouble[i], count);
+
+    	            if(count > max) {
+    	                max  = count;
+    	                mode = arrDouble[i];
+    	            }
+    	        }
+    	        else {
+    	            amode.put(arrDouble[i],1);
+    	            if(mode==0) {
+    	            	mode = arrDouble[i];
+    	            }
+    	        }
+    	    }
+
+            return (NodeValue.makeDecimal(mode));

```

Comment:
Extra ()

Reply:
change made.

Reply:
I see no change (nor for any of the other extra () ).

Reply:
ah I see, you have referred to the outer ( ) here as well.  change now in fork .